### PR TITLE
XSUP-41118 panorama bug fix

### PR DESCRIPTION
**Pull request by eyalpalo**
## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: [link to the issue](https://jira-dc.paloaltonetworks.com/browse/XSUP-41118)

## Description
Fixed an issue where Panorama incidents in the same time were not fetched.
## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

File: Packs/PAN-OS/Integrations/Panorama/Panorama.py [hunk 1]
```diff
@@ -14418,8 +14411,38 @@ def get_parsed_incident_entries(incident_entries_dict: dict[str, list[dict[str,
...
+                log_time = dateparser.parse(entry.get(time_field, ''), settings={'TIMEZONE': 'UTC'})
+
+                if (not log_type in offset_fetch_dict or
+                    not last_fetch_time or
+                    log_time > last_fetch_time):
+                    offset_fetch_dict[log_type] = 1  # type: ignore[literal-required]
+                    last_fetch_time = log_time
+                elif log_time == last_fetch_time:
+                    offset_fetch_dict[log_type] += 1  # type: ignore[literal-required]
+
```
Comment (jlevypaloalto) [inline Packs/PAN-OS/Integrations/Panorama/Panorama.py:14441]:
```suggestion
                else:
                    break
```

Comment (eyalpalo) [reply Packs/PAN-OS/Integrations/Panorama/Panorama.py:14868]: We assume that the incidents are ordered no? In that case if wont happen

File: Packs/PAN-OS/Integrations/Panorama/Panorama.py [hunk 2]
```diff
@@ -14844,8 +14838,38 @@ def get_parsed_incident_entries(incident_entries_dict: dict[str, list[dict[str,
...
+            last_fetch_dict (Dict[str, str]): last fetch dictionary
+            offset_fetch_dict (Dict[str, Dict]): last offset dictionary
+
+    """
+    for log_type, incident_entries in incident_entries_dict.items():
+        if incident_entries:
+            last_fetch_time = dateparser.parse(last_fetch_dict.get(log_type, ''), settings={'TIMEZONE': 'UTC'})  # type: ignore[literal-required]
+            for entry in incident_entries:
+                time_field = 'match_time' if log_type == 'Correlation' else 'time_generated'
+                log_time = dateparser.parse(entry.get(time_field, ''), settings={'TIMEZONE': 'UTC'})
```
Comment (dantavori) [inline Packs/PAN-OS/Integrations/Panorama/Panorama.py:14859]:
```suggestion
                if not (log_time := dateparser.parse(entry.get(time_field, ''), settings={'TIMEZONE': 'UTC'})):
                    raise DemistoException(f"{time_field=} is not a valid date in entry of {log_type=}.\n{entry=}")
```

File: Packs/PAN-OS/Integrations/Panorama/Panorama.py [hunk 3]
```diff
@@ -14937,8 +14964,7 @@ def main():  # pragma: no cover
             configured_max_fetch = arg_to_number(params['max_fetch'])
             queries = log_types_queries_to_dict(params)
             fetch_max_attempts = arg_to_number(params['fetch_job_polling_max_num_attempts'])
-            max_fetch = cast(MaxFetch, dict.fromkeys(queries, configured_max_fetch)
-                             | last_run.get('max_fetch_dict', {}))  # type: ignore[arg-type, operator]
+            max_fetch = cast(MaxFetch, dict.fromkeys(queries, configured_max_fetch))
```
Comment (dantavori) [inline Packs/PAN-OS/Integrations/Panorama/Panorama.py:14967]: where is the max_fetch used now? why is this change needed?

Comment (eyalpalo) [reply Packs/PAN-OS/Integrations/Panorama/Panorama.py:14967]: Talked about

**Review by dantavori — APPROVED** [review]
cool cool cool
